stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -2,30 +2,315 @@
+    }
+
+    @Override
+    public void onChildRemoved(Object parentElement, Object childElement) {
+      ChildNodeRemovedEvent message = new ChildNodeRemovedEvent();
+      message.parentNodeId = getOrCreateNodeIdForElement(parentElement);
+      message.nodeId = getOrCreateNodeIdForElement(childElement);
+      mPeerManager.sendNotificationToPeers("DOM.childNodeRemoved", message);
+      unhookElement(childElement);
+    }
+  }
+
+  private static class AttributeModifiedEvent {
+    @JsonProperty(required = true)
+    public int nodeId;
+
+    @JsonProperty(required = true)
+    public String name;
+
+    @JsonProperty(required = true)
+    public String value;
+  }
+
+  private static class AttributeRemovedEvent {
+    @JsonProperty(required = true)
+    public int nodeId;
+
+    @JsonProperty(required = true)
+    public String name;
+  }
+
+  private static class ChildNodeInsertedEvent {
+    @JsonProperty(required = true)
+    public int parentNodeId;
+
+    @JsonProperty(required = true)
+    public int previousNodeId;
+
+    @JsonProperty(required = true)
+    public Node node;
+  }
+
+  private static class ChildNodeRemovedEvent {
+    @JsonProperty(required = true)
+    public int parentNodeId;
+
+    @JsonProperty(required = true)
+    public int nodeId;
+  }
+
+  @ChromeDevtoolsMethod
+  public void enable(JsonRpcPeer peer, JSONObject params) {
+    mPeerManager.addPeer(peer);
+  }
+
+  @ChromeDevtoolsMethod
+  public void disable(JsonRpcPeer peer, JSONObject params) {
+    mPeerManager.removePeer(peer);
   }
 
   @ChromeDevtoolsMethod
   public JsonRpcResult getDocument(JsonRpcPeer peer, JSONObject params) {
-    Node node = new Node();
-    node.nodeId = 1;
-    node.nodeType = 1;
-    node.nodeName = "";
-    node.nodeValue = "";
-    return node;
+    Object rootElement = mDOMProvider.getRootElement();
+    if (rootElement == null) {
+      return null;
+    }
+
+    Node rootNode = createNodeForElement(rootElement);
+
+    GetDocumentResponse result = new GetDocumentResponse();
+    result.root = rootNode;
+    return result;
+  }
+
+  @ChromeDevtoolsMethod
+  public void highlightNode(JsonRpcPeer peer, JSONObject params) {
+    HighlightNodeRequest request = mObjectMapper.convertValue(params, HighlightNodeRequest.class);
+    if (request.nodeId != null) {
+      Object element = getElementForNodeId(request.nodeId);
+      // TODO: pass in the HighlightConfig
+      mDOMProvider.highlightElement(element);
+    }
   }
 
   @ChromeDevtoolsMethod
   public void hideHighlight(JsonRpcPeer peer, JSONObject params) {
+    mDOMProvider.hideHighlight();
+  }
+
+  public static class HighlightNodeRequest {
+    @JsonProperty(required = true)
+    public HighlightConfig highlightConfig;
+
+    @JsonProperty(required = false)
+    public Integer nodeId;
+
+    @JsonProperty(required = false)
+    public String objectId;
+  }
+
+  public static class HighlightConfig {
+    @JsonProperty(required = false)
+    public Boolean showInfo;
```

Comment:
The class or the field? HighlightConfig is part of the HighlightNode payload, and showInfo is part of it. We're not using it though ... can probably remove it? The old docs I was using has far fewer fields than the newer and more official docs we found